### ISSUE TITLE
Algorithm for Bot

### ISSUE THREAD

**Issue by maryjess** [maryjess/6nimmts#17]
### Would there be other possibilities to make a better 'opponent'?
So that the game has three level of bot difficulty:
* Easy
* Normal
* Hard
* (Extreme) Hard (?)

---

### Multiplayer?
#40

**Comment by maryjess** [comment]
# What "Bot" means: 
* 6nimmt itu bisa up to 10 players
* we need to find a way to make one bot
* soalnya kalo one bot jadi --> 10 bot bisa terjadi
* tinggal dinomerin bot 1, bot 2 dst

## Current way of how the bot works itu basic banget
iirc, cuman ada Easy + Normal difficulty currently
* Easy --> bruteforce (?)
* Normal --> (?) 

maryjess lupa botnya kerjanya gimana lol

**Comment by maryjess** [comment]
> yg udh 5 itu prioritas trakir
```
--> berarti kayak semacam bikin 'priority list' of the kebos currently in hand,
--> where kalo jumlah kebo >= 5 itu prioritas-nya burik
--> burik as in, bot-nya would not place those shitty cards

gitu?
```
---

Klo gw pikir, kartu yg kebo 5+ tu ditaro aj sih
Tapi pas ngambil, di avoid karna banyak kebo klo botnya pinter

**Comment by maryjess** [comment]
## Jadi pertanyaannya: ada cara biar si bot lebih "pinter" ga?
hence, game difficulty bisa pilih easy / normal / hard

```
* dasarnya tu botnya cari kartu dia yg selisihnya plg kecil ama salah satu angka di blakang, tapi yg udh 5 itu prioritas trakir
* Salah satu cara utk dia lebi "pinter" dikit; klo dia bakal harus ambil, dia bisa itung mana row yg kebo nya plg dikit
* Karna kn kartu masing2 jumlah kebo nya beda
* Soalnya kn tujuan 6nimmt tu minimise kebo yg kita punya, mayoritas kartu cuma ada 1 kebo, tapi bbrp punya lebih
```

**Comment by maryjess** [comment]
Even crazier, multiplayer? # SAVE THIS IDEA FOR LATER 🤯

**Comment by maryjess** [comment]
> yg udh 5 itu prioritas trakir
```
--> berarti kayak semacam bikin 'priority list' of the bulls currently in hand,
--> where kalo jumlah bull >= 5 itu prioritas-nya burik
--> burik as in, bot-nya would not place those cards with a lot of bulls

gitu?
```
---

Klo gw pikir, kartu yg kebo 5+ tu ditaro aj sih
Tapi pas ngambil, di avoid karna banyak kebo klo botnya pinter

**Comment by JSW-8180** [comment]
klo mau, bisa
Easy - normal bot, just put card with least difference against cards on the field
Normal - Ability utk liat row yg kebonya plg dikit klo harus ambil (and meibi prioritasin kluarin kartu di tangan dia yg kebonya banyak?)
Hard - Ability to fk over other players (klo mau extra jahat, priority ke playernya over other bots {if even possible}). Maybe bisa inget kartu yg diambil org laen, jadi dia bisa play kartu dri hand dia yg bakal bikin kita gabisa kluarin kartu kita (e.g. ada row yg 47 di field, kita punya 50, trus botnya taro kartu 52 di row itu, walopun dia punya kartu 36 yg bisa ditaro di row yg ujungnya 35)

Deep Blue mode (low priority, mostly for giggles) - Pake neural network utk ngetrain botnya biar bisa pinter as fk maen 6nimmt (? I don't completely understand neural network, ide darma lmao)

**Comment by JSW-8180** [comment]
> klo mau, bisa Easy - normal bot, just put card with least difference against cards on the field Normal - Ability utk liat row yg kebonya plg dikit klo harus ambil (and meibi prioritasin kluarin kartu di tangan dia yg kebonya banyak?) Hard - Ability to fk over other players (klo mau extra jahat, priority ke playernya over other bots {if even possible}). Maybe bisa inget kartu yg diambil org laen, jadi dia bisa play kartu dri hand dia yg bakal bikin kita gabisa kluarin kartu kita (e.g. ada row yg 47 di field, kita punya 50, trus botnya taro kartu 52 di row itu, walopun dia punya kartu 36 yg bisa ditaro di row yg ujungnya 35)
> 
> Deep Blue mode (low priority, mostly for giggles) - Pake neural network utk ngetrain botnya biar bisa pinter as fk maen 6nimmt (? I don't completely understand neural network, ide darma lmao)

obviously, gw gtau python coding, jadi klo gk feasible, gk usah diikutin xD

**Comment by maryjess** [comment]
```
=== EASY ===
normal bot, just put card with least difference against cards on the field

=== NORMAL ===
Ability utk liat row yg kebonya plg dikit klo harus ambil 
(and meibi prioritasin kluarin kartu di tangan dia yg kebonya banyak?)

=== HARD ===
Ability to 📛 over other players
(klo mau extra jahat, priority ke playernya over other bots {if even possible}). 
Maybe bisa inget kartu yg diambil org laen, 
jadi dia bisa play kartu dri hand dia yg bakal bikin kita gabisa kluarin kartu kita 
(e.g. ada row yg 47 di field, kita punya 50, trus botnya taro kartu 52 di row itu, 
walopun dia punya kartu 36 yg bisa ditaro di row yg ujungnya 35)

=== DEEP BLUE? HAHAHA ===
Deep Blue mode 
(low priority, mostly for giggles)
Pake neural network utk ngetrain botnya biar bisa pinter as 📛 maen 6nimmt 
(? I don't completely understand neural network, ide @CizBargah  lmao)
```

>  I don't completely understand neural network

Its ok bro i also dont HAHAHAHA
